include/swift/AST/DiagnosticGroups.h

```diff
@@ -24,16 +24,25 @@
 #include <unordered_map>
 
 namespace swift {
+enum class DiagnosticGroupOptions {
+  /// No options.
+  none,
+
+  /// The diagnostic warnings belonging to this group should be ignored by default,
+  /// but will be re-enabled by various warning options (-Wwarning, -Werror).
+  DefaultIgnoreWarnings,
```

Comment:
out of curiosity, is functionality like this expected to remain 'one way'? i.e. you can go from ignored to not, but you can't go back.

Reply:
That's right; but, we're building towards a way to go in the other direction with source-level controls:
https://forums.swift.org/t/pitch-source-level-control-over-compiler-warnings/